src/messages/SharedMessageBuilder.cpp

```diff
@@ -30,6 +30,34 @@ namespace {
         }
     }
 
+    QStringList parseTagList(const QVariantMap &tags, const QString &key)
+    {
+        auto iterator = tags.find(key);
+        if (iterator == tags.end())
+            return QStringList{};
+
+        return iterator.value().toString().split(
+            ',', QString::SplitBehavior::SkipEmptyParts);
+    }
+
+    std::vector<Badge> parseBadges(const QVariantMap &tags)
```

Comment:
Duplicate of TwitchMessageBuilder's `parseBadges` - fine to keep duplicate for now